flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/PlannerQueryOperation.java

```diff
@@ -72,6 +77,15 @@ public String asSummaryString() {
                 "PlannerNode", Collections.emptyMap(), getChildren(), Operation::asSummaryString);
     }
 
+    @Override
+    public String asSerializableString() {
+        try {
+            return toSqlString.get();
+        } catch (Exception e) {
+            throw new TableException("Could not convert Calcite tree to a SQL string", e);
```

Comment:
I would rephrase to `Given plan is not serializable into SQL` or so. Not mention Calicte.